In [3]:
import pandas as pd

# Load the IPL data into a pandas dataframe

In [82]:
data = pd.read_csv('C:\\Users\\Bala Vignesh.A\\Desktop\\SCMA 632\\Cricket_data.csv')


In [6]:
print(data)

      season       id                                          name  \
0        NaN  1370350          Chennai Super Kings v Gujarat Titans   
1        NaN  1370351         Lucknow Super Giants v Mumbai Indians   
2        NaN  1370352                                     TBC v TBC   
3        NaN  1370353                                     TBC v TBC   
4     2023.0  1359544  Royal Challengers Bangalore v Gujarat Titans   
...      ...      ...                                           ...   
1027  2008.0   336012  Royal Challengers Bangalore v Mumbai Indians   
1028  2008.0   336019            Kings XI Punjab v Rajasthan Royals   
1029  2008.0   336038           Delhi Daredevils v Rajasthan Royals   
1030  2008.0   336039         Chennai Super Kings v Kings XI Punjab   
1031  2008.0   336040        Chennai Super Kings v Rajasthan Royals   

      short_name                                        description home_team  \
0       CSK v GT  Qualifier 1 (N), Indian Premier League at Chen..

In [17]:
print(ct_data)

Empty DataFrame
Columns: [season, id, name, short_name, description, home_team, away_team, toss_won, decision, 1st_inning_score, 2nd_inning_score, winner, result, start_date, end_date, venue_id, venue_name, home_captain, away_captain, pom, points, super_over, home_overs, home_runs, home_wickets, home_boundaries, away_overs, away_runs, away_wickets, away_boundaries, highlights, home_key_batsman, home_key_bowler, home_playx1, away_playx1, away_key_batsman, away_key_bowler, match_days, umpire1, umpire2, tv_umpire, referee, reserve_umpire]
Index: []

[0 rows x 43 columns]


In [18]:
(data['home_key_batsman'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_key_batsman, Length: 1032, dtype: bool

In [27]:
(data['home_key_bowler'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_key_bowler, Length: 1032, dtype: bool

In [28]:
(data['home_playx1'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_playx1, Length: 1032, dtype: bool

In [29]:
(data['away_playx1'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_playx1, Length: 1032, dtype: bool

In [30]:
(data['away_key_batsman'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_key_batsman, Length: 1032, dtype: bool

In [31]:
 (data['away_key_bowler'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_key_bowler, Length: 1032, dtype: bool

In [32]:
columns_to_check = ['home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler']

filter_mask = data[columns_to_check].apply(lambda x: x.str.contains('Rahul Tewatia')).any(axis=1)

rahul_tewatia_data = data[filter_mask]

print(rahul_tewatia_data)

     season       id                                            name  \
4    2023.0  1359544    Royal Challengers Bangalore v Gujarat Titans   
12   2023.0  1359536            Gujarat Titans v Sunrisers Hyderabad   
17   2023.0  1359531                 Mumbai Indians v Gujarat Titans   
23   2023.0  1359525           Gujarat Titans v Lucknow Super Giants   
26   2023.0  1359522               Rajasthan Royals v Gujarat Titans   
..      ...      ...                                             ...   
525  2015.0   829741              Rajasthan Royals v Kings XI Punjab   
532  2015.0   829755        Kolkata Knight Riders v Rajasthan Royals   
592  2014.0   733979        Rajasthan Royals v Kolkata Knight Riders   
602  2014.0   733999  Royal Challengers Bangalore v Rajasthan Royals   
619  2014.0   734033              Kings XI Punjab v Rajasthan Royals   

    short_name                                        description home_team  \
4     RCB v GT  70th Match (N), Indian Premier League at

In [35]:
# Select only the rows that match the filter criteria
rahul_tewatia_alone_data = data[filter_mask]

# Count the number of rows where 'Rahul Tewatia' appears alone
num_alone = rahul_tewatia_alone_data.shape[0]

print(f"Number of rows where 'Rahul Tewatia' appears alone: {num_alone}")

Number of rows where 'Rahul Tewatia' appears alone: 79


In [38]:
columns_to_check = ['home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler']

In [39]:
filter_mask = data[columns_to_check].apply(lambda x: x.str.count('Rahul Tewatia') == 1).any(axis=1)

In [41]:
print(data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


In [42]:
data.columns = data.columns.str.strip()

In [43]:
print(data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


# Assuming 'data' is the DataFrame containing the IPL data

### Create a new DataFrame with the required columns

In [50]:
ipl_data = data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'away_overs', 'away_runs', 'away_wickets', 'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']]


In [51]:
print(ipl_data)

      season       id                                          name  \
0        NaN  1370350          Chennai Super Kings v Gujarat Titans   
1        NaN  1370351         Lucknow Super Giants v Mumbai Indians   
2        NaN  1370352                                     TBC v TBC   
3        NaN  1370353                                     TBC v TBC   
4     2023.0  1359544  Royal Challengers Bangalore v Gujarat Titans   
...      ...      ...                                           ...   
1027  2008.0   336012  Royal Challengers Bangalore v Mumbai Indians   
1028  2008.0   336019            Kings XI Punjab v Rajasthan Royals   
1029  2008.0   336038           Delhi Daredevils v Rajasthan Royals   
1030  2008.0   336039         Chennai Super Kings v Kings XI Punjab   
1031  2008.0   336040        Chennai Super Kings v Rajasthan Royals   

      short_name                                        description home_team  \
0       CSK v GT  Qualifier 1 (N), Indian Premier League at Chen..

# Extract the batsman, ball, runs, and wickets per player per match

In [52]:
ipl_data_batting = ipl_data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'home_key_batsman', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']].rename(columns={'home_key_batsman': 'Player'})
ipl_data_bowling = ipl_data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'home_key_bowler', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']].rename(columns={'home_key_bowler': 'Player'})
ipl_data_batting['Runs'] = ipl_data_batting['home_runs']
ipl_data_bowling['Wickets'] = ipl_data_bowling['home_wickets']
ipl_data_batting['Balls'] = ipl_data_batting['home_overs'] * 6
ipl_data_bowling['Balls'] = ipl_data_bowling['home_overs'] * 6

# Combine the batting and bowling data

In [53]:
ipl_data_player = pd.concat([ipl_data_batting, ipl_data_bowling], axis=0)

# Group by IPL round, player, and player type, and calculate total runs and wickets

In [54]:
ipl_data_grouped = ipl_data_player.groupby(['season', 'Player']).agg({'Runs': 'sum', 'Wickets': 'sum', 'Balls': 'sum'}).reset_index()

# Find the top three run-getters and top three wicket-takers in each IPL round

In [56]:
top_run_getters = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Runs')).reset_index(drop=True)
top_wicket_takers = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Wickets')).reset_index(drop=True)

# Print the results

In [57]:
print("Top Run-Getters in Each IPL Round:")
print(top_run_getters)
print("\nTop Wicket-Takers in Each IPL Round:")
print(top_wicket_takers)

Top Run-Getters in Each IPL Round:
    season                           Player   Runs  Wickets  Balls
0   2008.0    Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1   2008.0     Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2   2008.0    Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2
3   2009.0          Matthew Hayden,MS Dhoni  482.0      0.0  342.6
4   2009.0        Yuvraj Singh,Irfan Pathan  291.0      0.0  240.0
5   2009.0      AB de Villiers,David Warner  243.0      0.0  213.6
6   2010.0     Ambati Rayudu,Saurabh Tiwary  390.0      0.0  240.0
7   2010.0        Murali Vijay,Albie Morkel  246.0      0.0  120.0
8   2010.0     Jacques Kallis,Robin Uthappa  204.0      0.0  111.0
9   2011.0      Michael Hussey,Murali Vijay  325.0      0.0  240.0
10  2011.0   Ambati Rayudu,Sachin Tendulkar  283.0      0.0  240.0
11  2011.0       Shaun Marsh,Dinesh Karthik  282.0      0.0  240.0
12  2012.0  Gautam Gambhir,Brendon McCullum  498.0      0.0  360.0
13  2012.0          Faf du 

In [58]:
import scipy.stats as stats
import matplotlib.pyplot as plt

# Extract the top three batsmen and bowlers in the last three IPL tournaments

In [66]:
print("Top Run-Getters in Each IPL Round:")


Top Run-Getters in Each IPL Round:


In [67]:
print(top_run_getters)

    season                           Player   Runs  Wickets  Balls
0   2008.0    Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1   2008.0     Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2   2008.0    Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2
3   2009.0          Matthew Hayden,MS Dhoni  482.0      0.0  342.6
4   2009.0        Yuvraj Singh,Irfan Pathan  291.0      0.0  240.0
5   2009.0      AB de Villiers,David Warner  243.0      0.0  213.6
6   2010.0     Ambati Rayudu,Saurabh Tiwary  390.0      0.0  240.0
7   2010.0        Murali Vijay,Albie Morkel  246.0      0.0  120.0
8   2010.0     Jacques Kallis,Robin Uthappa  204.0      0.0  111.0
9   2011.0      Michael Hussey,Murali Vijay  325.0      0.0  240.0
10  2011.0   Ambati Rayudu,Sachin Tendulkar  283.0      0.0  240.0
11  2011.0       Shaun Marsh,Dinesh Karthik  282.0      0.0  240.0
12  2012.0  Gautam Gambhir,Brendon McCullum  498.0      0.0  360.0
13  2012.0          Faf du Plessis,MS Dhoni  368.0      0.0  2

In [69]:
print(top_run_getters.head(3))

   season                         Player   Runs  Wickets  Balls
0  2008.0  Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1  2008.0   Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2  2008.0  Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2


In [72]:
print("\nTop Wicket-Takers in Each IPL Round:")


Top Wicket-Takers in Each IPL Round:


In [73]:
print(top_wicket_takers)

    season                                Player  Runs  Wickets  Balls
0   2008.0             Praveen Kumar,Zaheer Khan   0.0     18.0  235.2
1   2008.0          Farveez Maharoof,Amit Mishra   0.0     10.0   96.6
2   2008.0            Piyush Chawla,Irfan Pathan   0.0     10.0  120.0
3   2009.0            Anil Kumble,Jacques Kallis   0.0     17.0  236.4
4   2009.0      Lakshmipathy Balaji,Albie Morkel   0.0     14.0  240.0
5   2009.0                RP Singh,Venugopal Rao   0.0     13.0  240.0
6   2010.0           Zaheer Khan,Harbhajan Singh   0.0     17.0  349.8
7   2010.0          Doug Bollinger,Shadab Jakati   0.0     16.0  240.0
8   2010.0                 RP Singh,Pragyan Ojha   0.0     16.0  224.4
9   2011.0             Rahul Sharma,Yuvraj Singh   0.0     16.0  240.0
10  2011.0                Johan Botha,Amit Singh   0.0     13.0  224.4
11  2011.0            Munaf Patel,Lasith Malinga   0.0     11.0  235.2
12  2012.0                Dale Steyn,Amit Mishra   0.0     14.0  230.4
13  20

In [74]:
print(top_wicket_takers.head(3))

   season                        Player  Runs  Wickets  Balls
0  2008.0     Praveen Kumar,Zaheer Khan   0.0     18.0  235.2
1  2008.0  Farveez Maharoof,Amit Mishra   0.0     10.0   96.6
2  2008.0    Piyush Chawla,Irfan Pathan   0.0     10.0  120.0


# Calculate the performance metrics (e.g., runs, wickets, boundaries)


In [76]:
print(rahul_tewatia_data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


In [79]:
rahul_tewatia_performance = rahul_tewatia_data.groupby('season')[['home_runs', 'home_wickets', 'home_boundaries', 'away_runs', 'away_wickets', 'away_boundaries']].sum()

# Calculate the average performance metrics

In [80]:
average_performance = rahul_tewatia_performance.mean()

# Print the relationship between performance and salary

In [81]:
print("Relationship between Rahul Tewatia's performance and salary:")
print("Salary: 9 crore")
print("Average runs: ", average_performance['home_runs'] + average_performance['away_runs'])
print("Average wickets: ", average_performance['home_wickets'] + average_performance['away_wickets'])
print("Average boundaries: ", average_performance['home_boundaries'] + average_performance['away_boundaries'])

Relationship between Rahul Tewatia's performance and salary:
Salary: 9 crore
Average runs:  2878.777777777778
Average wickets:  101.11111111111111
Average boundaries:  359.1111111111111


###This output shows the average performance of Rahul Tewatia in matches where his salary is 9 crore. The average runs scored by Tewatia in these matches is approximately 2878.78, the average wickets taken is approximately 101.11, and the average boundaries hit is approximately 359.11.

This information can help in understanding the relationship between Tewatia's performance and his salary. If his performance is consistently high, it could indicate that he is being paid a fair salary. However, if his performance is low compared to his salary, it could suggest that he is overpaid. Further analysis would be needed to make a definitive conclusion.